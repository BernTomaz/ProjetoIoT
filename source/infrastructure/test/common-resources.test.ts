// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import "@aws-cdk/assert/jest";
import { SynthUtils } from "@aws-cdk/assert";
import { CommonResourcesConstruct}  from "../lib/common-resources";
import { Stack } from "aws-cdk-lib";

test('IoT Device Simulator CommonResourceConstruct Test', () => {
    const stack = new Stack();

    const commonResources = new CommonResourcesConstruct(stack, 'TestCommonResource', {
        sourceCodeBucket: "test-bucket"
    });

    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
