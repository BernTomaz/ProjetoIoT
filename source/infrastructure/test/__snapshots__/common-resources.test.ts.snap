// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IoT Device Simulator CommonResourceConstruct Test 1`] = `
{
  "Resources": {
    "TestCommonResourceLogBucket0C3C89D7": {
      "DeletionPolicy": "Retain",
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W35",
              "reason": "This bucket is to store S3 logs, so it does not require access logs.",
            },
            {
              "id": "W51",
              "reason": "This bucket is to store S3 logs, so it does not require S3 policy.",
            },
          ],
        },
      },
      "Properties": {
        "AccessControl": "LogDeliveryWrite",
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "OwnershipControls": {
          "Rules": [
            {
              "ObjectOwnership": "ObjectWriter",
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
